expfmt/text_parse.go

```diff
@@ -190,28 +195,38 @@ func (p *TextParser) startComment() stateFn {
 	if p.readTokenAsMetricName(); p.err != nil {
 		return nil // Unexpected end of input.
 	}
-	if p.currentByte == '\n' {
+	p.setOrCreateCurrentMF()
+
```

Comment:
The general coding style in this file is to avoid vertical whitespace. Could you keep it that way?